src/include/sof/audio/component.h

```diff
@@ -165,13 +165,24 @@ enum {
 #define comp_cl_dbg(drv_p, __e, ...) LOG_DBG(__e, ##__VA_ARGS__)
 
 /* device level tracing */
-#define comp_err(comp_p, __e, ...) LOG_ERR(__e, ##__VA_ARGS__)
 
-#define comp_warn(comp_p, __e, ...) LOG_WRN(__e, ##__VA_ARGS__)
+#if CONFIG_IPC_MAJOR_4
+#define __COMP_FMT "comp:%u.%x "
```

Comment:
The component ID is always between 0x10000 and 0xfffff i.e. it always has 5 hexadecimal digits, right? If it might become 0x110000 or something else, maybe we want to fix the width?

Reply:
possible to translate the ID into a name(like PGA) in the log? or the ID is fixed, we can use this ID to directly know which module it is?

Reply:
the log context will tell you what component is logging.

> pipe: pipe:1.0 pipe trigger cmd 2

The first pipe is the log context.

Reply:
@btian1 The names are not part of the binary (sof-logger does the translation on host side in the tool -> we could have similar postprocessing of the log file that looks up the components, but this is not a trivial task, especially to deploy as you will need access to the ELF binary when parsing).

Reply:
> es are n

thanks, let's do this in parse script.

Reply:
@lyakh I don't get the 5-digit limitation. It's a bitmask of 16bit module id (upper 16 bits) and 16bit instance id (lower bits). I could not find any place that would limit this to to "low value" module ids (albeit current topologies use low-number module ids).